cli/src/complete.rs

```diff
@@ -448,14 +448,27 @@ pub fn leaf_config_keys() -> Vec<CompletionCandidate> {
     config_keys_impl(true)
 }
 
-fn all_files_from_rev(rev: String) -> Vec<CompletionCandidate> {
+fn dir_prefix_from<'a>(path: &'a str, current: &str) -> Option<&'a str> {
+    path.strip_prefix(current)?
```

Comment:
nit: just find separator from `current.len()`? Callers should exclude paths not under the current path.